src/aggregator/aggregator/generic_elem.go

```diff
@@ -103,26 +103,37 @@ type typeSpecificElemBase interface {
 	Close()
 }
 
-type lockedAggregation struct {
-	sync.Mutex
+//nolint:maligned
+type timedAggregation struct {
```

Comment:
nit: If there's some state you want to keep all covered by one mutex you can use the pattern we sometimes use of:
```go
type someStruct struct {
  foo int
  bar int

  readWrittenState *someStructReadWrittenState
  otherState *someStructOtherState
}

type someStructReadWrittenState struct {
  sync.Mutex // guards this state fields
  baz int
  qux int
}

type someStructOtherState struct {
  sync.Mutex // guards this state fields
  qaz int
  bux int
}
```

Reply:
I guess the tricky part is now we are explicitly reading stuff guarded by the mutex without holding the mutex (because another mutex is acquired instead).